proposed-html-structure-from-wordpress.php

```diff
@@ -0,0 +1,185 @@
+<?php
+
+class Research_Guide {
+	private $taxonomies;
+	private $categories;
+	private $tags;
+	private $guidance;
+	private $url;
+	private $title;
+	private $slug;
+	private $id;
+	private $recommended;
+	private $partners;
+	private $online;
+
+	public function __construct($post) {
+		$this->set_url($post);
+		$this->set_title($post);
+		$this->set_slug($post);
+		$this->set_id($post);
+		$this->set_taxonomies($post);
+		$this->set_categories($post);
+		$this->set_tags($post);
+		$this->set_guidance($post);
+		$this->set_recommended();
+		$this->set_online();
+		$this->set_partners();
+	}
+
+	function set_url ($post) {
+		$this->url = get_permalink($post);
+	}
+	function set_title ($post) {
+		$this->title = get_the_title($post);
+	}
+	function set_slug ($post) {
+		$this->slug = $post->post_name;
+	}
+	function set_id ($post) {
+		$this->id = $post->ID;
+	}
+	function set_taxonomies ($post) {
+		$this->taxonomies = get_the_taxonomies($post);
+	}
+	//only run after set_taxonomies()
+	function set_categories ($post) {
+		if ($this->taxonomies == null) {
+			$this->set_taxonomies($post);
+		}  else {
+			$this->categories = strip_tags($this->taxonomies['category']);
+		}
+	}
+	function set_tags ($post) {
+		if ($this->taxonomies == null) {
+			$this->set_taxonomies($post);
+		} else {
+			$this->tags = strip_tags( $this->taxonomies['post_tag'] );
+		}
+	}
+	function set_guidance ($post) {
+		if ($this->taxonomies == null) {
+			$this->set_taxonomies($post);
+		} else {
+			$this->guidance = strip_tags($this->taxonomies['guidance']);
+		}
+	}
+	//can only run after set_guidance()
+	function set_recommended () {
+		if (!strpos($this->guidance, 'Recommended')) {
+			$this->recommended = 'false';
+		} else {
+			$this->recommended = 'true';
+		}
+	}
+	function set_online () {
+			if (!strpos($this->guidance, 'All online')) {
+			$this->online = 'false';
+		} else {
+			$this->online = 'true';
+		}
+	}
+	//can only run after set_categories()
+	function set_partners () {
+		if (strpos($this->categories, 'Findmypast.co.uk') !== false || strpos($this->categories, 'Ancestry.co.uk') !== false ) {
```

Comment:
I'm not sure this is an ideal candidate for an `if` condition. I'd suggest having an array of partners and iterating over them to identify whether they're a sub-string of `$this->categories`. If any partner is found set `$this->partners` to a string representing the name of the partner (see proposed-html-structure.html). Because you'll need both the name as it's stored in WordPress (i.e. Findmypast.co.uk) and the class name (i.e. 'find-my-past') this is a candidate for an associative array. Let me know if you need any help with this.